sites/website/src/docs/design-systems/fast-frame.md

```diff
@@ -4,228 +4,297 @@ title: The FAST Frame Design System
+j
+For all APIs described in this section, the configuration overrides apply to all components registered to the DesignSystem except when the option is explicitly provided [during component registration](/docs/design-systems/fast-frame#configuring-components-during-registration). 
+
+#### `DesignSystem.withPrefix()`
+As [Web Components](https://developer.mozilla.org/en-US/docs/Web/Web_Components), FAST Components are [Custom Elements](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements) subject to the [naming conventions defined by HTML](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name). This means that the element name must contain a ‘-‘ character. By default, the DesignSystem defines all elements registered with it with the “fast” prefix. [The example above](/docs/design-systems/fast-frame/#add-element-to-html) illustrates this, where the FastButton is defined with the ‘fast’ prefix.
+
+This prefix can be changed for all components registered by the DesignSystem using the `DesignSystem.withPrefix()` API:
 
-**Example: Using the [`FASTDesignSystemProvider`](/docs/api/fast-components.fastdesignsystemprovider)**
 ```ts
-import { FASTDesignSystemProvider } from "@microsoft/fast-components";
+import { DesignSystem } from "@microsoft/fast-foundation";
+import { fastButton } from "@microsoft/fast-components";
+
+DesignSystem
+    .getOrCreate()
+    .withPrefix("faster")
+    .register(fastButton())
 
-FASTDesignSystemProvider; // Prevent tree shaking
 ```
+In this case, the element can be used in HTML using the ‘faster’:
 ```html
-<fast-design-system-provider use-defaults>
-    Hello world
-</fast-design-system-provider>
+<faster-button>Click me!</faster-button>
+```
+
+#### `DesignSystem.withShadowRootMode()`
+Unless specified during [component registration creation](/docs/design-systems/creating-a-component-library#compose-and-export-registration), all components registered with the DesignSystem are defined with an [open shadow root mode.](https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot/mode). This behavior can be changed using `DesignSystem.withShadowRootMode()` to close all shadow roots by default:
+
+```ts
+DesignSystem
+    .getOrCreate()
+    .withShadowRootMode("closed")
+    .register(/* ... */)
+```
+
+#### `DesignSystem.withElementDisambiguation()`
+If two elements registered with the DesignSystem share the same tag name, the name can be disambiguated using `DesignSystem.withElementDisambiguation()`, allowing opportunity to provide a new name if there are conflicts:
+
+```ts
+DesignSystem
+    .getOrCreate()
+    .withElementDisambiguation((name: string, type: Constructable<{}>) => {
+        return `disambiguated-${name}`;
+    })
+    .register(/* ... */
+```
```

Comment:
I think I understand quite well why this is needed, but I could not understand how I'm supposed to use this in real life, until I read the section ```What is the `ElementDefinitionContext`?```  which demos

```ts
const template = (context: ElementDefinitionContext) => {
	const buttonTag = context.tagFor(FASTButton);
	return html`<${buttonTag}>Click me!</${buttonTag}>`
}
```

and yet I'm not sure I get all of this right
maybe link this section to the other one and give a full picture